docs/tutorials/mongoid-queries.txt

```diff
@@ -967,14 +978,69 @@ recommended to use the block form described above:
 .. code-block:: ruby
 
   Mongoid::QueryCache.enabled = true
-  
+
   # ...
-  
+
   Mongoid::QueryCache.enabled = false
 
 Mongoid also provides a :ref:`Rack middleware <query-cache-middleware>`
 to enable the query cache automatically for each web request.
 
+The Improved Driver Query Cache
+```````````````````````````````
+
+The Mongoid query cache has been reimplemented in version 2.14.0 of the MongoDB
+Ruby Driver. The driver query cache is more correct and more effective, and it
+is recmomended that you upgrade to version 2.14.0 of the Ruby driver or newer.
+
+With driver versions 2.14.0 or newer, Mongoid will use the driver query cache
+instead of its own. When used with older versions of the driver (that do not
+implement the query cache) Mongoid will fall back on the legacy query cache.
+
+Mongoid will retain the interface (described above) for enabling and disabling
+the query cache. When using driver versions 2.14.0 or newer, this interface
+will affect the driver query cache. 
+
+Read more about the Ruby driver query cache `in the driver documentation
+<https://docs.mongodb.com/ruby-driver/current/tutorials/query-cache/>`_.
+
+Legacy Query Cache Limitations
+``````````````````````````````
+
+The following is a list of limitations of the Mongoid query cache:
+
+* The query cache does not cache query results that exceed the batch size.
+  The default batch size is 1000 documents.
+
+* The query cache does not take into account read preference or read concern
+  when deciding whether to return cached results. When performing the same
+  query with a different read preference or read concern with the query cache
+  enabled, the driver may return incorrect results.
```

Comment:
s/driver/Mongoid/ ? Or I would phrase it as "incorrect results may be returned" to avoid explicitly mentioning either one.